### DISCUSSION TITLE
CallbackRouteError with discord.

### DISCUSSION THREAD

**Discussion by 1k3w** [nextauthjs/next-auth#6323] [Help]
Hi! i am using sveltekit and am using authjs. though, when following the docs and changing the github example to Discord it breaks with this error:
```challenge {
  scheme: 'basic',
  parameters: { error: 'invalid_client', error_description: '' }
}
CallbackRouteError: TODO: Handle www-authenticate challenges as needed
    at Module.callback (file:///W:/PB/sveltekit-panel/node_modules/@auth/core/lib/routes/callback.js:243:23)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async AuthInternal (file:///W:/PB/sveltekit-panel/node_modules/@auth/core/lib/index.js:52:38)
    at async Proxy.Auth (file:///W:/PB/sveltekit-panel/node_modules/@auth/core/index.js:95:30)
    at async respond (file:///W:/PB/sveltekit-panel/node_modules/@sveltejs/kit/src/runtime/server/index.js:229:20)
    at async file:///W:/PB/sveltekit-panel/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:447:22
[auth][error][CallbackRouteError]: Read more at https://errors.authjs.dev#callbackrouteerror
{
  "provider": "discord"
}```

this is my code:

```
import { SvelteKitAuth } from "@auth/sveltekit"
import Discord from "@auth/core/providers/discord"
import { redirect, type Handle } from '@sveltejs/kit';
import { sequence } from '@sveltejs/kit/hooks';

async function authorization({ event, resolve }) {
	// Protect any routes other then /auth
	if (!event.url.pathname.startsWith('/auth')) {
		const session = await event.locals.getSession();
		if (!session) {
			throw redirect(303, '/auth');
		}
	}

	// If the request is still here, just proceed as normally
	const result = await resolve(event, {
		transformPageChunk: ({ html }) => html
	});
	return result;
}

export const handle: Handle = sequence(SvelteKitAuth({
	providers: [Discord({
		clientId: '<redacted>', clientSecret: '<redacted>', 
 authorization: "https://discord.com/api/oauth2/authorize?scope=identify+email+guilds"
	})],
}),
	authorization)
```

I have looked at the link the error provided me, but i havent gotten any wiser off of it.

**Comment by wngasinur** [comment]
I'm getting the same issue. Any solution for this ?
```
challenge { scheme: 'basic', parameters: { realm: 'PF AS Token Endpoints' } }
[auth][error][CallbackRouteError]: Read more at https://errors.authjs.dev#callbackrouteerror
[auth][cause]: Error: TODO: Handle www-authenticate challenges as needed
    at handleOAuth (file:///C:/workspace/code/svelte-app/node_modules/@auth/sveltekit/node_modules/@auth/core/lib/oauth/callback.js:74:15)
```

**Comment by xpluscal** [comment]
Same issue here. Any update on that?

**Comment by caseybaggz** [comment]
Same here but using Solid Start. 😢

**Reply by thomporter** [reply]
+1 ;(

**Comment by jasonHooten** [comment]
No updates in 10 months, that's disappointing.

**Comment by nicolaeser** [comment]
I am still getting that error since i reinstalled my pc. Never saw it before.

**Comment by rodrgds** [comment]
still no solution for this? do I just drop authjs?

**Comment by chrimesdev** [comment]
Also getting this error using a third party OIDC

**Comment by paukraft** [comment]
Hi everyone,

I encountered a similar error in my project:

```
challenge {
  scheme: 'basic',
  parameters: { error: 'invalid_client', error_description: '' }
}
[auth][error] CallbackRouteError: Read more at https://errors.authjs.dev#callbackrouteerror
[auth][cause]: Error: TODO: Handle www-authenticate challenges as needed
```

In my case, the issue was due to a typo in my `.env` file—I had accidentally added an 's' at the end of my secret key. After correcting my `ID` and `SECRET` values, the error was resolved.

I recommend double-checking your environment variables to ensure they are correctly specified. This might help fix the issue for you as well.

Hope this helps!
